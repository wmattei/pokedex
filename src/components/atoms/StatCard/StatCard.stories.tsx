import { text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import * as React from 'react';
import StatCard from '.';
import CenterView from '../../../../storybook/stories/CenterView';

storiesOf('StatCard', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .addDecorator(withKnobs)
  .add('Card com informações numéricas', () => (
    <StatCard label={text('StatLabel', 'HP')} value={text('StatValue', '12')} />
  ));
