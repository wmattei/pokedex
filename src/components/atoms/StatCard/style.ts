import { StyleSheet } from 'react-native';
import { Theme } from 'react-native-paper/lib/typescript/types';

export default (theme: Theme) =>
  StyleSheet.create({
    container: {
      width: 100,
      height: 100,
      borderColor: theme.colors.primary,
      alignItems: 'center',
      justifyContent: 'space-evenly',
      borderWidth: 1,
      borderStyle: 'solid',
      borderRadius: 5,
    },
    value: {
      fontSize: 30,
      color: theme.colors.primary,
    },
  });
