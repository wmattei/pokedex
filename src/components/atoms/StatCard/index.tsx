import * as React from 'react';
import { Text, View } from 'react-native';
import { useTheme } from 'react-native-paper';
import style from './style';

type Props = {
  label: string;
  value: string | number;
};

export default function StatCard({ label, value }: Props) {
  const theme = useTheme();
  const themedStyle = style(theme);

  return (
    <View style={themedStyle.container}>
      <Text style={themedStyle.value}>{value}</Text>
      <Text>{label}</Text>
    </View>
  );
}
