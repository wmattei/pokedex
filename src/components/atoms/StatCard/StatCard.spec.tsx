import * as React from 'react';
import { create } from 'react-test-renderer';
import StatCard from '.';

describe('StatCard', () => {
  it('Renders correctly', () => {
    const tree = create(<StatCard label="HP" value={4} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
