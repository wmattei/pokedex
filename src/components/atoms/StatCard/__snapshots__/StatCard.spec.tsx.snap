// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatCard Renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#6200ee",
      "borderRadius": 5,
      "borderStyle": "solid",
      "borderWidth": 1,
      "height": 100,
      "justifyContent": "space-evenly",
      "width": 100,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#6200ee",
        "fontSize": 30,
      }
    }
  >
    4
  </Text>
  <Text>
    HP
  </Text>
</View>
`;
